src/Extension/Propagator/Instana/InstanaMultiPropagator.php

```diff
@@ -0,0 +1,175 @@
+    {
+        $getter ??= ArrayAccessGetterSetter::getInstance();
+        $context ??= Context::getCurrent();
+
+        $traceId = self::readHeader($carrier, $getter, self::INSTANA_TRACE_ID_HEADER);
+        $spanId = self::readHeader($carrier, $getter, self::INSTANA_SPAN_ID_HEADER);
+        $level = self::getSampledValue($carrier, $getter);
+
+        $spanContext = self::extractImpl($carrier, $getter);
+
+        if (($traceId === '' &&  $spanId === '') && $level !== null) {
+            return (new NonRecordingSpan($spanContext))
+                ->storeInContext($context);
+
+        } elseif (!$spanContext->isValid()) {
+            return $context;
+        }
+
+        return $context->withContextValue(Span::wrap($spanContext));
+
+    }
+
+    private static function readHeader($carrier, PropagationGetterInterface $getter, string $key): string
+    {
+        // By convention, X-INSTANA-* headers are sent all-upper-case. For http and http/2, Node.js normalizes all headers to
+        // all-lower-case, that's why we first read via the lower case variant of the header name. For other protocols, no
+        // such normalization happens, so we also need to read via the original all-upper-case variant.
+        $header = $getter->get($carrier, strtolower($key)) ?: $getter->get($carrier, $key);
+
+        // Return the header or an empty string if not found
+        return $header ?: '';
+    }
+    private static function getSampledValue($carrier, PropagationGetterInterface $getter): ?int
+    {
+        $value = $getter->get($carrier, self::INSTANA_LEVEL_HEADER);
+
+        if ($value === null) {
+            return null;
+        }
+
+        if (in_array(strtolower($value), self::VALID_SAMPLED)) {
+            return (int) self::IS_SAMPLED;
+        }
+        if (in_array(strtolower($value), self::VALID_NON_SAMPLED)) {
+            return (int) self::IS_NOT_SAMPLED;
+        }
+
+        return null;
+    }
+
+    private static function extractImpl($carrier, PropagationGetterInterface $getter): SpanContextInterface
+    {
+        $traceId = self::readHeader($carrier, $getter, self::INSTANA_TRACE_ID_HEADER);
+        $spanId = self::readHeader($carrier, $getter, self::INSTANA_SPAN_ID_HEADER);
+        $level = self::getSampledValue($carrier, $getter);
+
+        if ($traceId && strlen($traceId) < 32) {
+            $traceId = str_pad($traceId, 32, '0', STR_PAD_LEFT);
+        }
+
+        if ($spanId && strlen($spanId) < 16) {
+            $spanId =  str_pad($spanId, 16, '0', STR_PAD_LEFT);
+        }
+
+        if ((SpanContextValidator::isValidTraceId($traceId) || SpanContextValidator::isValidSpanId($spanId))) {
+            return SpanContext::createFromRemoteParent(
+                $traceId,
+                $spanId,
+                $level ? TraceFlags::SAMPLED : TraceFlags::DEFAULT
+            );
+        }
```

Comment:
```suggestion
        if ((SpanContextValidator::isValidTraceId($traceId) && SpanContextValidator::isValidSpanId($spanId))) {
            return SpanContext::createFromRemoteParent(
                $traceId,
                $spanId,
                $level ? TraceFlags::SAMPLED : TraceFlags::DEFAULT
            );
        }
```

Shouldn't this condition be true only when both `traceId` and `spanId` are valid?